### DISCUSSION TITLE
Auto require plugin

### DISCUSSION THREAD

**Discussion by Willem-J-an** [lumen-oss/nvim-best-practices#17] [Q&A] [answered]
I followed some of the instructions in here for my new plugin. I did the vim.g config table, and created a single command with autocompletion. Now I'm wondering about this, provided I have a lua/myplugin.lua which creates my magnificent autocompletion command, is it okay for me to have a plugin/init.lua file which requires('myplugin')? My plugin lazy loads everything when required as suggested in this repo. 

Having the auto self require will make the command directly available to the users, so they don't have to require my plugin in their config. Curious about the best practice on this point.

**Comment by mrcjkb** [comment, marked as answer]
Hey 👋 

> Having the auto self require will make the command directly available to the users, so they don't have to require my plugin in their config.

Yep, I would recommend that (see the [initialisation section](https://github.com/nvim-neorocks/nvim-best-practices?tab=readme-ov-file#x-dont-4)).

It can be a good idea to create the commands in the `plugin/` script, in such a way that nothing needs to be `require`d during initialisation.

**Reply by Willem-J-an** [reply]
Now that I read it again it finally clicks. So I move the command creation to the plugin script, and when the user executed the command it will lazily require my module!
